### DISCUSSION TITLE
PrettyGCode 2

### DISCUSSION THREAD

**Discussion by Kragrathea** [Kragrathea/OctoPrint-PrettyGCode#128] [General]
I am working on a new version that is a substantial rewrite from the previous version. The code for it (for now) lives in the TestBranch and can be installed with this link

https://github.com/Kragrathea/OctoPrint-PrettyGCode/archive/refs/heads/TestBranch.zip

You can force the plugin version to use the old PrettyGCode for testing. Add the parameter "?oldpgc=1" to the url.

In addition to showing up as a OctoPrint plugin it will also work as a stand alone web application. Make sure you are logged into Octoprint and then browse to <octoprint server>/plugin/prettygcode/static/pgcode/pgcode.html

Notes:
- The bed volume isn't being detected currently.
- Camera view isn't working yet
- Perspective camera option requires manual reload of the page.

**Comment by jneilliii** [comment]
I think there are issues with center origin in this newer version.
new:
![image](https://user-images.githubusercontent.com/5249455/134428458-46ecd677-54db-446d-afb1-d9ef15256256.png)
old:
![image](https://user-images.githubusercontent.com/5249455/134428656-5aade1c4-a8ab-4b71-b186-300030c96eab.png)

rendering seems slower than previous version too.

**Comment by Kragrathea** [comment]
Oh, I forgot to mention. The bed volume isn't being detected currently. 

If you want to modify the code to fix it temp you can change pgcframe.js line 112 to be like this:
``
$("#pgciframe").attr("src","/plugin/prettygcode/static/pgcode/pgcode.html?embedded=1&bed.origin=center&bed.width=400&bed.depth=400&server="+document.location.origin)
``

**Reply by jneilliii** [reply]
Yeah, much better. I just used that command in developer console like this.

```
$("#pgciframe").attr("src","/plugin/prettygcode/static/pgcode/pgcode.html?embedded=1&bed.origin=center&bed.width=200&bed.depth=200&server="+document.location.origin);
```

![image](https://user-images.githubusercontent.com/5249455/134431232-e2c734d3-8ff6-4781-bda9-c0af202a617b.png)

**Reply by Kragrathea** [reply]
Nice! You can also add &fps=1 to get a frame rate number on screen. 

The only reason I can think of that it would render slower is because you have travel lines on. Those were not drawn in the last version.

**Comment by Kragrathea** [comment]
Also looks like the layer value might be wrong. And the font isn't exactly what it is is supposed to be.

**Comment by jneilliii** [comment]
rendering seems a little closer to the original when I turn off travel moves and the 2d view.

**Reply by Kragrathea** [reply]
Right! The 2d view could take a lot of frame rate. Although it is only rendering the currently layer so it shouldn't be like 2x slower.

**Comment by DARKFiB3R** [comment]
Hi Kragrathea, is  PrettyGCode 2 still being worked on?

Thank you for the current version, it's a really great plugin.

**Reply by niko2020** [reply]
Waiting for V2!

**Comment by trebory6** [comment]
Also wanted to ask if PrettyGCode 2 is being worked on. 

It looks like it's been over a year since Kragathea last commented on this at all, and the ReadMe.md was edited also over a year ago. I hope they're ok, because it's not normal for someone to just completely abandon a project like this.

I was hoping a newer version was being worked on, because I wanted to change the theming for a "retro" Gcode viewer that uses this framework but makes it look like a retrofuturistic scifi display instead.

Sad that this looks like a dead project. 

Edit: Looks like Kragathea's made a contribution in Jul 2022 on another project, so at least they're not dead. Hope everything else is ok though. Wish we could get the tiniest update on the status.

**Reply by jneilliii** [reply]
> it's not normal for someone to just completely abandon a project like this.

actually pretty common in the plugin world it seems. 

https://plugins.octoprint.org/abandoned/

**Comment by Kragrathea** [comment]
Hi. PrettyCode 2 is pretty much the pgcode project here:
https://github.com/Kragrathea/pgcode
Pgcode will work with in the future will work directly with OctoPrint but it is not as a plugin. It will need to installed on the Pi separately.

This it the OctoPrint Plugin version and I may come back and update it at some point.

**Comment by Z1mDMan** [comment]
> I am working on a new version that is a substantial rewrite from the previous version. The code for it (for now) lives in the TestBranch and can be installed with this link
> 
> https://github.com/Kragrathea/OctoPrint-PrettyGCode/archive/refs/heads/TestBranch.zip
> 
> You can force the plugin version to use the old PrettyGCode for testing. Add the parameter "?oldpgc=1" to the url.
> 
> In addition to showing up as a OctoPrint plugin it will also work as a stand alone web application. Make sure you are logged into Octoprint and then browse to /plugin/prettygcode/static/pgcode/pgcode.html
> 
> Notes:
> 
> * The bed volume isn't being detected currently.
> * Camera view isn't working yet
> * Perspective camera option requires manual reload of the page.

been having some issues lately with the generally available version so thought I would try this and I actually kind of like it. I really hope developement continues. Pretty Gcode is the best plugin out there.
